Entity/Player/Player.gd

```diff
@@ -6,7 +6,12 @@ class_name Player
 func _ready() -> void:
 	super()
 	PlayerManager.set_player(self)
+	get_energy_component().init_entity_component(self)
```

Comment:
is this needed ? I know that for health we don't need to do it since it's not in the code anywhere, why do you need to do it for the energy component ?

Reply:
if it's redundant it should be removed